File: codeforces/practice/818B/game.scala
import scala.io.StdIn._

object Main {
  def main(args: Array[String]): Unit = {
    val nm = readLine.split(" ").map(_.toInt)
    val n = nm.head
    val m = nm.last
    val arr = readLine.split(" ").map(_.toInt)
    val visit = new Array[Int](n)
    val result = new Array[Int](n)
    var possible = true
    arr.zipWithIndex.foreach { k =>
      val index = k._2
      if (index > 0) {
        val diff = (k._1 - arr(index - 1) + n)%n
        val rDiff = if (diff == 0) n else diff
        if (visit(rDiff - 1) > 0 && visit(rDiff - 1) != arr(index - 1)) possible = false
        result(arr(index - 1) - 1) = rDiff
        visit(rDiff - 1) = arr(index - 1)
      }
    }
    // Fill remaining with unvisited
    val available = result.zipWithIndex.filter(x => x._1 == 0).map(_._2)
    val unvisited = visit.zipWithIndex.filter(x =>  x._1 == 0).map(_._2)
    if (available.size != unvisited.size) possible = false
    (available zip unvisited).foreach { pair =>
      result(pair._1) = pair._2 + 1
    }
    if (possible == false) println(-1) else println(result.mkString(" "))
  }
}

File: codeforces/practice/818B/game.cpp
#include <bits/stdc++.h>

using namespace std;

int arr[101];
int res[101];
int seen[101];
int main() {
  int n, m;
  int possible = 0;
  cin >> n >> m;
  for(int i=0;i<m;i++) cin >> arr[i];
  for(int i=1;i<m;i++) {
    int diff = (arr[i] - arr[i - 1] + n)%n;
    int rDiff = diff == 0? n : diff;
    if (seen[rDiff] > 0 && (seen[rDiff] != arr[i - 1] )) possible = 1;
    res[arr[i - 1]] = rDiff;
    seen[rDiff] = arr[i - 1];
  }
  int u = 1;
  while((u <=n) && (seen[u] != 0)) u++;
  int k;
  for(k=1; k<=n; k++) {
    if (res[k] == 0) {
      // Try to get a valid missing number here
      if (u > n) {
        possible = 1;
        break;
      }
      res[k] = u;
      seen[u] = k;
      while((u <= n) && (seen[u] != 0)) u++;
    }
  }
  if (possible == 0) {
    for(int p=1; p<=n; p++) {
      printf("%d ", res[p]);
    }
    printf("\n");
  } else {
    printf("%d\n", -1);
  }
  return 0;
}
